### DISCUSSION TITLE
new layer w GeomText, nudge_x and y inherited.

### DISCUSSION THREAD

**Discussion by EvaMaeRey** [ggplot2-extenders/ggplot-extension-club#6] [General]
I've noticed that nudge_x and nudge_y doesn't seem to be available when using new stat + GeomText, which I've found surprising. 
Issues warning about unknown parameter nudge_x.   Is it meant to be so? 

library(tidyverse)

compute_group_xy_means <- function(data, scales){
  
  data |>
    summarise(x = mean(x),
              y = mean(y),
              label = paste0("(", round(x),",", (round(y)), ")"))
  
}

StatXymeans <- ggproto("StatXymeans", 
                       Stat,
                       compute_group = compute_group_xy_means)

ggplot(cars) +
  aes(speed, dist) + 
  geom_point() + 
  layer(stat = StatXymeans,
        geom = GeomText,
        position = "identity",
        params = list(nudge_x = .2) 
        )

#> Warning: Ignoring unknown parameters: `nudge_x`

Created on 2024-04-05 with [reprex v2.0.2](https://reprex.tidyverse.org/)

**Comment by teunbrand** [comment]
The `nudge_x` and `nudge_y` arguments to `geom_text()` are shortcuts in parameterising a `position_nudge()` for the layer. The function body if you print the function shows how they're used. 

> Is it meant to be so?

Yes the `GeomText` class does not recognise the nudge parameters.

**Comment by EvaMaeRey** [comment]
Thank you for the explanation!  That makes sense.   While thinking about this, I wondered about a different hacky solution, which would be to cross geom_text and geom_label: Label paddinng for GeomText, which would give spacing in npc which is sometimes nicer.  But, I think you can't get the text look w/ padding w the available functions though.  

``` r
library(tidyverse)

# use geom label as text - making label disappear
ggplot(cars) + 
  aes(speed, dist) + 
  geom_point() +
  geom_label(label = "Point", 
             label.padding = unit(0.2, "lines"),
             vjust = 1, 
             hjust = 0,
             alpha = 0, # remove background
             linewidth = 0, # not available
             check_overlap = T)
#> Warning in geom_label(label = "Point", label.padding = unit(0.2, "lines"), :
#> Ignoring unknown parameters: `linewidth` and `check_overlap`
```

![](https://i.imgur.com/NNtLOdR.png)

``` r


# nice to have text padding
ggplot(cars) + 
  aes(speed, dist) + 
  geom_point() +
  geom_text(label = "Point", 
            label.padding = unit(0.2, "lines"), # nice to have...
            vjust = 1, 
            hjust = 0,
            check_overlap = T)
#> Warning in geom_text(label = "Point", label.padding = unit(0.2, "lines"), :
#> Ignoring unknown parameters: `label.padding`
```

![](https://i.imgur.com/sZWzvcr.png)

<sup>Created on 2024-04-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by teunbrand** [comment]
The linewidth of the label is in the `label.size` parameter, so you can hide it that way.
See also: https://github.com/teunbrand/ggplot_tricks?tab=readme-ov-file#labels-as-text

**Comment by EvaMaeRey** [comment]
Ah, stellar!  So many excellent nuggets in ggplot tricks!

``` r
library(ggplot2)

# check_overlap ... that this argument is not supported by geom_label()
ggplot(cars) + 
  aes(speed, dist) + 
  geom_point() +
  geom_label(label = "Point", 
             label.padding = unit(0.2, "lines"),
             vjust = 1, 
             hjust = 0,
             alpha = 0, # remove background
             label.size = 0) # remove border
```

![](https://i.imgur.com/vw4pPFo.png)

<sup>Created on 2024-04-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by EvaMaeRey** [comment]
Looking more at ggplot tricks, for example in the text-contrasting-fill section (very cool work), I kind prefer a + aes_colour_autocontrast_after_fill method rather than aes(     ,    , !!!aes_colour_autocontrast_after_fill), like this:
![image](https://github.com/teunbrand/ggplot-extension-club/assets/10491910/def1326e-bdd9-4ef4-959e-e20e4e7f6c55)
